numpy_array_api_compat/_aliases.py

```diff
@@ -158,148 +207,192 @@ def asarray(
     *,
     dtype: Optional[Dtype] = None,
     device: Optional[Device] = None,
-    copy: Optional[Union[bool, np._CopyMode]] = None,
+    copy: "Optional[Union[bool, np._CopyMode]]" = None,
+    namespace = None,
 ) -> ndarray:
+    if namespace is None:
+        try:
+            xp = get_namespace(obj, _use_compat=False)
+        except ValueError:
+            # TODO: What about lists of arrays?
```

Comment:
What you have seems reasonable. Libraries should be passing `namespace=numpy` if they want this to work, or otherwise handle lists/sequences themselves.